### PR TITLE
Support multi-process Firefox

### PR DESCRIPTION
**Pull request by lydell**
This PR tracks the progress of fixing #378.

### CODE REVIEW

File: extension/lib/commands-frame.coffee [hunk 1]
```diff
@@ -0,0 +1,325 @@
...
+utils = require('./utils')
+
+{ isProperLink, isTextInputElement, isContentEditable } = utils
+
+XULDocument = Ci.nsIDOMXULDocument
+
+storage  = {}
+commands = {}
+
+commands.go_up_path = ({ content, count }) ->
```
Comment (lydell) [inline extension/lib/commands-frame.coffee]: Need to remove `content` being passed in here (and in a few other places).

**Comment by akhodakivskiy** [comment]
This looks kick ass!

**Comment by lydell** [comment]
I’ve made some great progress here this weekend. All the architecture has been figured out. What’s left is mostly one really large debug session.

**Comment by LordJZ** [comment]
> I’ve made some great progress here this weekend. All the architecture has been figured out. What’s left is mostly one really large debug session.

:+1: Thanks for your hard work!

**Comment by macat** [comment]
Exciting PR!

**Comment by lydell** [comment]
Some more progress now. VimFx now boots without errors, adds the toolbar button, has the options interface and the help dialog working, and many commands work. However, many still don’t and there are a few bugs. But definitely a big step in the right direction!

**Comment by lydell** [comment]
Things start getting stable now.

TODO: Counts with several digits are broken.

**Comment by lydell** [comment]
TODO:

- Fix custom `<tab>` handling in text inputs
- Fix broken tests
- Document `vim`

**Comment by lydell** [comment]
I’m going to merge this PR later today, even though there are some minor quirks left to be solved. I want to work on a few different things of VimFx, and not merging this makes it difficult.

**Comment by mozillazg** [comment]
:+1:

**Comment by lydell** [comment]
I’ve gone through every command now, and they all seem to work. I’ve also surfed around for a while without noticing anything strange or any errors in the error console. I’ll merge this now and minor bugs/regressions (if any) will be fixed as we find them.

**Comment by quartz55** [comment]
Has this been tested with "dom.ipc.processCount" > 1?

**Comment by lydell** [comment]
No. What is that?

**Comment by quartz55** [comment]
It's basicly how many plugin-container processes firefox spawns.

I'm not sure if that's really the problem, but I've been having issues with multiple firefox windows and focus. For example, when I have 2 windows and I have window 2 selected, vimfx shortcuts control window 1.

Will test a bit more see if that's really the case :+1:

**Comment by lydell** [comment]
Yes, there have been problems with multiple Firefox windows, but I fixed them today. Try the latest develop.

**Pull request merged**